freqtrade/persistence/models.py

```diff
@@ -278,6 +297,66 @@ def open_date_utc(self):
     def close_date_utc(self):
         return self.close_date.replace(tzinfo=timezone.utc)
 
+    @property
+    def enter_side(self) -> str:
+        if self.is_short:
+            return "sell"
+        else:
+            return "buy"
+
+    @property
+    def exit_side(self) -> str:
+        if self.is_short:
+            return "buy"
+        else:
+            return "sell"
+
+    def __init__(self, **kwargs):
+        for key in kwargs:
+            setattr(self, key, kwargs[key])
+        if self.liquidation_price:
+            self.set_liquidation_price(self.liquidation_price)
+        self.recalc_open_trade_value()
+
+    def set_stop_loss(self, stop_loss: float):
+        """
+            Method you should use to set self.stop_loss.
+            Assures stop_loss is not passed the liquidation price
+        """
+        if self.liquidation_price is not None:
+            if self.is_short:
+                sl = min(stop_loss, self.liquidation_price)
+            else:
+                sl = max(stop_loss, self.liquidation_price)
+        else:
+            sl = stop_loss
+
+        if not self.stop_loss:
+            self.initial_stop_loss = sl
+        self.stop_loss = sl
+
+    def set_liquidation_price(self, liquidation_price: float):
+        """
+            Method you should use to set self.liquidation price.
+            Assures stop_loss is not passed the liquidation price
+        """
+        if self.stop_loss is not None:
+            if self.is_short:
+                self.stop_loss = min(self.stop_loss, liquidation_price)
+            else:
+                self.stop_loss = max(self.stop_loss, liquidation_price)
+        else:
+            self.initial_stop_loss = liquidation_price
+            self.stop_loss = liquidation_price
+
+        self.liquidation_price = liquidation_price
+
+    def __repr__(self):
+        open_since = self.open_date.strftime(DATETIME_PRINT_FORMAT) if self.is_open else 'closed'
+
+        return (f'Trade(id={self.id}, pair={self.pair}, amount={self.amount:.8f}, '
```

Comment:
I think this should also include the side (maybe also leverage?).

Reply:
In another PR, that will change a lot of tests

Reply:
By side do you mean short/long ? It already prints buy/sell